### PR TITLE
removing unnecessary limitation of scope to maximum 1 parameter

### PR DESCRIPTION
**Pull request by klobuczek**
Fixes #

This pull introduces/changes:
 * scope can now take more than 1 parameter
 * the previous error message was very confusing if defined scope with 2 parameters and called with 2:
` wrong number of arguments (given 0, expected 1)`

Pings:
@cheerfulstoic
@subvertallchris

### CODE REVIEW

File: spec/e2e/scope_spec.rb [hunk 1]
```diff
@@ -97,6 +97,26 @@
     end
   end
 
+  describe 'Person.scope :great_students, -> (identifier, score) { where("#{identifier}.score > ?", score)' do
+    before(:each) do
+      Person.scope :great_students, -> (identifier, score) { where("#{identifier}.score > ?", score || 41) }
```
Comment (ProGM) [inline spec/e2e/scope_spec.rb]: This lambda syntax is not valid on jruby 1.7 https://github.com/jruby/jruby/issues/2704

It should just work by removing the empty space between the `->` and `(identifier, scope) ...`:

```ruby
Person.scope :great_students, ->(identifier, score) { where("#{identifier}.score > ?", score || 41) }
```

**Comment by ProGM** [comment]
I noticed this problem as well, this is great :+1: 
I left a comment on it

File: lib/neo4j/active_node/scope.rb [hunk 2]
```diff
@@ -68,14 +68,9 @@ def full_scopes
       end
 
       def _call_scope_context(eval_context, query_params, proc)
-        if proc.arity == 1
-          eval_context.instance_exec(query_params, &proc)
-        else
-          eval_context.instance_exec(&proc)
-        end
+        eval_context.instance_exec(*query_params.fill(nil, query_params.length..proc.arity - 1), &proc)
```
Comment (cheerfulstoic) [inline lib/neo4j/active_node/scope.rb]: Ah, nice, I didn't know about `Array#fill`!

:+1:

File: spec/e2e/scope_spec.rb [hunk 3]
```diff
@@ -97,6 +97,26 @@
     end
   end
 
+  describe 'Person.scope :great_students, -> (identifier, score) { where("#{identifier}.score > ?", score)' do
+    before(:each) do
+      Person.scope :great_students, ->(identifier, score) { where("#{identifier}.score > ?", score || 41) }
```
Comment (cheerfulstoic) [inline spec/e2e/scope_spec.rb]: This is fine, though I think that in 8.0 we support calling `identity` from within a scope now

Comment (klobuczek) [inline spec/e2e/scope_spec.rb]: I was merely looking for a test with 2 parameters. :)

Comment (cheerfulstoic) [inline spec/e2e/scope_spec.rb]: I thought so ;)  Just wanted to make sure that you knew about it__

**Pull request merged**